pkg/dataplane/streamconsumergroup/streamconsumergroup.go

```diff
@@ -86,7 +86,7 @@ func (scg *streamConsumerGroup) getTotalNumberOfShards() (int, error) {
 	return response.Output.(*v3io.DescribeStreamOutput).ShardCount, nil
 }
 
-func (scg *streamConsumerGroup) setState(modifier stateModifier) (*State, error) {
+func (scg *streamConsumerGroup) setState(modifier stateModifier, postSetStateInPersistencyModifier func() error) (*State, error) {
```

Comment:
move func signature to a type (same as `stateModifier`).
we do that to ease testings by allowing providing custom callback by any need